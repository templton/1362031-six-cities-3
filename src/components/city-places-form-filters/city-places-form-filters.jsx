import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {sortCityFilterType, sortPlacesInCity} from "../../actions";
import CityPlacesFormFiltersItem from "../city-places-form-filters-item/city-places-form-filters-item";

const CityPlacesFormFilters = (props) => {
  const {onFilterClick, currentPlaceFilterType, isFilterVisible, onSortPlacesInCity} = props;
  const handleFilterClick = (filterType) => {
    onSortPlacesInCity(filterType);
    onFilterClick();
  };
  return (
    isFilterVisible &&
    <ul className="places__options places__options--custom places__options--opened">
      {
        Object.values(sortCityFilterType).map((item) => {
          return (
            <CityPlacesFormFiltersItem
              key={item}
              onClick={handleFilterClick}
              className={`places__option ${currentPlaceFilterType === item ? `places__option--active` : ``}`}
              item={item}
            />
          );
        })
      }
    </ul>
  );
};

CityPlacesFormFilters.propTypes = {
  onFilterClick: PropTypes.func.isRequired,
  onSortPlacesInCity: PropTypes.func.isRequired,
  currentPlaceFilterType: PropTypes.string.isRequired,
  isFilterVisible: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  currentPlaceFilterType: null
});

const mapDispatchToProps = (dispatch) => ({
  onSortPlacesInCity: (filterType) => dispatch(sortPlacesInCity(filterType))
});

export {CityPlacesFormFilters};
export default connect(mapStateToProps, mapDispatchToProps)(CityPlacesFormFilters);
