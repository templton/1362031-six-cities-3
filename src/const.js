export const PlaceCardTypes = {
  APARTMENT: `apartment`,
  ROOM: `room`,
  HOUSE: `house`,
  HOTEL: `hotel`,
};

export const sortCityFilterType = {
  POPULAR: `Popular`,
  TOP_RATED_FIRST: `Top rated first`,
  PRICE_LOW_TO_HIGH: `Price: low to high`,
  PRICE_HIGH_TO_LOW: `Price: high to low`,
};

export const routerPath = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVOURITES: `/favorites`,
  HOTEL_DETAIL: `/hotel/:hotelId`
};
