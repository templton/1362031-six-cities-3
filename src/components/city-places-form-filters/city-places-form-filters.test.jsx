import React from "react";
import renderer from "react-test-renderer";
import {CityPlacesFormFilters} from "./city-places-form-filters";
import {sortCityFilterType} from "../../const";

describe(`CityPlacesFormFilters render`, () => {
  it(`simple render`, ()=>{
    const tree = renderer
      .create(<CityPlacesFormFilters
        onFilterClick={()=>{}}
        onSortPlacesInCity={()=>{}}
        onFilterChange={()=>{}}
        currentPlaceFilterType={sortCityFilterType.POPULAR}
        isFilterVisible={true}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
