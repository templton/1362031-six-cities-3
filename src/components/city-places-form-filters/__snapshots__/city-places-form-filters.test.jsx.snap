// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityPlacesFormFilters render simple render 1`] = `
<ul
  className="places__options places__options--custom places__options--opened"
>
  <li
    className="places__option places__option--active"
    onClick={[Function]}
  >
    Popular
  </li>
  <li
    className="places__option "
    onClick={[Function]}
  >
    Top rated first
  </li>
  <li
    className="places__option "
    onClick={[Function]}
  >
    Price: low to high
  </li>
  <li
    className="places__option "
    onClick={[Function]}
  >
    Price: high to low
  </li>
</ul>
`;
